querybook/server/lib/query_executor/executor_factory.py

```diff
@@ -36,6 +43,7 @@ def _get_executor_params_and_engine(query_execution_id, celery_task, session=Non
         raise ArchivedQueryEngine("This query engine is disabled.")
 
     client_setting = get_client_setting_from_engine(engine, uid, session=session)
+    client_setting["execution_type"] = execution_type
```

Comment:
i don't think ClientBaseClass needs to know `execution_type`, it is more of a adapter for different dbapi. execution_type can be passed to QueryExecutor and it can determine what flags to pass to query client

Reply:
this `client_setting` will pass to `QueryExecutorBaseClass._client_setting`, and it will use it to create query client
```    
def _get_cursor(self):
        if self._client is None:
            self._client = self._get_client(self._client_setting)

        return self._client.cursor()
```

or are you talking another path to pass it down?